File: xiniu/xiniu.js
var _keyStr = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
, _p = "W5D80NFZHAYB8EUI2T649RT2MNRMVE2O";

function e1(e) {
  if (null == e)
      return null;
  for (var t, n, r, o, i, a, s, u = "", c = 0; c < e.length; )
      o = (t = e.charCodeAt(c++)) >> 2,
      i = (3 & t) << 4 | (n = e.charCodeAt(c++)) >> 4,
      a = (15 & n) << 2 | (r = e.charCodeAt(c++)) >> 6,
      s = 63 & r,
      isNaN(n) ? a = s = 64 : isNaN(r) && (s = 64),
      u = u + _keyStr.charAt(o) + _keyStr.charAt(i) + _keyStr.charAt(a) + _keyStr.charAt(s);
  return u
}
function e2(e) {
  if (null == (e = _u_e(e)))
      return null;
  for (var t = "", n = 0; n < e.length; n++) {
      var r = _p.charCodeAt(n % _p.length);
      t += String.fromCharCode(e.charCodeAt(n) ^ r)
  }
  return t
}
function sig(e) {
  return md5(e + _p).toUpperCase()
}
function d1(e) {
    var t, n, r, o, i, a, s = "", u = 0;
    for (e = e.replace(/[^A-Za-z0-9\+\/\=]/g, ""); u < e.length; )
        t = _keyStr.indexOf(e.charAt(u++)) << 2 | (o = _keyStr.indexOf(e.charAt(u++))) >> 4,
            n = (15 & o) << 4 | (i = _keyStr.indexOf(e.charAt(u++))) >> 2,
            r = (3 & i) << 6 | (a = _keyStr.indexOf(e.charAt(u++))),
            s += String.fromCharCode(t),
        64 != i && (s += String.fromCharCode(n)),
        64 != a && (s += String.fromCharCode(r));
    return s
}
function _u_e(e) {
  if (null == e)
      return null;
  e = e.replace(/\r\n/g, "\n");
  for (var t = "", n = 0; n < e.length; n++) {
      var r = e.charCodeAt(n);
      r < 128 ? t += String.fromCharCode(r) : r > 127 && r < 2048 ? (t += String.fromCharCode(r >> 6 | 192),
      t += String.fromCharCode(63 & r | 128)) : (t += String.fromCharCode(r >> 12 | 224),
      t += String.fromCharCode(r >> 6 & 63 | 128),
      t += String.fromCharCode(63 & r | 128))
  }
  return t
}
function _u_d(e) {
  for (var t = "", n = 0, r = 0, o = 0, i = 0; n < e.length; )
      (r = e.charCodeAt(n)) < 128 ? (t += String.fromCharCode(r),
      n++) : r > 191 && r < 224 ? (o = e.charCodeAt(n + 1),
      t += String.fromCharCode((31 & r) << 6 | 63 & o),
      n += 2) : (o = e.charCodeAt(n + 1),
      i = e.charCodeAt(n + 2),
      t += String.fromCharCode((15 & r) << 12 | (63 & o) << 6 | 63 & i),
      n += 3);
  return t
}
function d2(e) {
  for (var t = "", n = 0; n < e.length; n++) {
      var r = _p.charCodeAt(n % _p.length);
      t += String.fromCharCode(e.charCodeAt(n) ^ r)
  }
  return t = _u_d(t)
}

function encrypt(c) {
    var l = JSON.parse(c);
    var f = e1(e2(JSON.stringify(l.payload)));
    //  , p = sig(f), md5(f + _p).toUpperCase() 请求参数里sig，在python中实现
    return f;
}
function decrypt(s) {
    var d = d1(s), 
    m = d2(d), 
    v = JSON.parse(m);
    return m;
}

File: xiniu/xiniu.py
import requests
import json
from hashlib import md5
import execjs

headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36',
    'Content-Type': 'application/json',
}

def sig(s):
    md5_v = str(md5(s.encode()).hexdigest()).upper()
    return md5_v

def de_cry(d):
    with open('xiniu.js', 'r',encoding='utf-8') as f:
        decrypt_js = f.read()
    ctx = execjs.compile(decrypt_js)
    data = ctx.call('decrypt', d)
    return data

def en_cry(n):
    with open('xiniu.js', 'r',encoding='utf-8') as f:
        encrypt_js = f.read()
    ctx = execjs.compile(encrypt_js)
    data = ctx.call('encrypt', n)
    return data

def company_lib():
    url = 'https://www.xiniudata.com/api/search2/company/lib'
    data = {
        "payload":"LBcgWUQrZGATHHVgSiogJ1Z2DG9kfnZeIi0zOT8qXG1tbhkUEjonPWp7Ah8UZywsXjhZQxtoD29hbCsoOildOAhBJV8SdB0HZGM9LVUgJj1bNxQOGyYmRyhsfm8wMFwrPlsjZ1QvMj9qewIfFGc8J0IhQhYDcHYebyEgKTM3EHV1USFLU2xqeDsuKzYaf2J/AmQCGBsiNVUobGh9emdBJi1QZgICfjs=",
        "sig":"DD2BC59355FD050B5850814DC002C693",
        "v":1
    }
    resp = requests.post(url, headers = headers, data = json.dumps(data)).json()
    res_json1 = de_cry(resp['d'])
    res_dic = json.loads(res_json1)
    list = []
    for i in res_dic['data']:
        list.append(i['companyCode'])
    dict = {"payload": {"codes": list}}
    str1 = str(json.dumps(dict))
    payload = en_cry(str1)
    return payload

def list_by_codes(payload):
    url = 'https://www.xiniudata.com/api2/service/x_service/person_company4_list/list_companies4_list_by_codes'
    p = "W5D80NFZHAYB8EUI2T649RT2MNRMVE2O"
    data1 = {"payload":payload,"sig":sig(payload + p),"v":1}
    resp = requests.post(url, headers = headers, data = json.dumps(data1)).json()
    res_json2 = de_cry(resp['d'])
    print(res_json2)

if __name__ == "__main__":
    payload = company_lib()
    list_by_codes(payload)